Lib/http/client.py

```diff
@@ -321,7 +321,7 @@ def begin(self):
 
         if self.debuglevel > 0:
             for hdr in self.headers:
```

Comment:
You could iterate `self.headers.items()`.

Reply:
+1.  This is actually the correct way to do this, as it accounts for instances where a header was sent multiple times with different values.  This chosen approach behaves incorrectly today, only printing whichever value is returned from `self.headers.get(hdr)` for each occurrence of `hdr`.

For example, if my response contained these headers:

```
x-goog-hash: crc32c=KAwGng==
x-goog-hash: md5=eB5eJF1ptWaXm4bijSPyxw==
```

The debug output prints this:

```
header: x-goog-hash: crc32c=KAwGng==
header: x-goog-hash: crc32c=KAwGng==
```

Reply:
Do you mind to create a PR for this change?

Reply:
Done, sent #12681.